src/utils/time.ts

```diff
@@ -27,24 +29,57 @@ function parse24HTime(time: string) {
     return time.split(':').map((x) => parseInt(x));
 }
 
-function compareTime(a: number[], b: number[]) {
-    if (a[0] === b[0] && a[1] === b[1]) {
+function compareTime(time1: number[], time2: number[]) {
+    if (time1[0] === time2[0] && time1[1] === time2[1]) {
         return 0;
     }
-    if (a[0] < b[0] || (a[0] === b[0] && a[1] < b[1])) {
+    if (time1[0] < time2[0] || (time1[0] === time2[0] && time1[1] < time2[1])) {
         return -1;
     }
     return 1;
 }
 
-export function isInTimeInterval(date: Date, time0: string, time1: string) {
+// a <= b
+function nextIntervalTime(time1, time2, currentTime, date: Date): number {
+    if (compareTime(time1, time2) === 0) {
+        return null;
+    }
+
+    if (compareTime(currentTime, time1) < 0) {
+        // t < a <= b
+        // Schedule for todate at time a
+        date.setHours(time1[0]);
+        date.setMinutes(time1[1]);
+        return date.getTime();
+    }
+
+    if (compareTime(currentTime, time2) < 0) {
+        // a <= t < b
+        // Schedule for today at time b
+        date.setHours(time2[0]);
+        date.setMinutes(time2[1]);
+        return date.getTime();
+    }
+
+    // a <= b <= t
+    // Schedule for tomorrow at time a
+    return (new Date(date.getFullYear(), date.getMonth(), date.getDate() + 1, time1[0], time1[1])).getTime();
+}
+
+export function isInTimeInterval(time0: string, time1: string, date: Date = new Date()): TimeCheck {
```

Comment:
`isInTimeInterval` should only return a boolean value, it should not know anything about next intervals etc.

Try exporting `nextIntervalTime` function, but make its name understandable out of context.

Reply:
Makes sense, will prepare a PR today or tomorrow.

Reply:
PR is at #6405.